### PR TITLE
ci: expand performance testing to catch errors, fixed connection refused errors, added performance testing with state.

### PR DESCRIPTION
**Pull request by gzukel**
# Description

1. Expanded the logic the pipeline to look for errors in the report json if no failures we're reported via artillery as a secondary check to ensure we know when tests fail/succeed.
2. Fixed connection refused errors. These errors we're happening locally. The issue is using the composite for artillery uses docker backed composite so the network was separate. The reports weren't clear that it was failing or if it was reporting wrong. That is corrected now.
3. Added starting a local net from the state export to performance test with state.
4. Added slack channel notification when testing fails so the DevOps team can know about it when it happens. 

# How Has This Been Tested?

I executed multiple Github Action scenarios to ensure all logic functioned as intended through failure and success scenarios.

Failed Test: https://github.com/zeta-chain/node/actions/runs/9719615016
Failed Test w/State: https://github.com/zeta-chain/node/actions/runs/9719839293
Success Test: https://github.com/zeta-chain/node/actions/runs/9719810694
Success Tests With State: https://github.com/zeta-chain/node/actions/runs/9719842212/job/26830323773

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [x] Tested via GitHub Actions

### CODE REVIEW

**Review by lumtis — APPROVED** [review]
LGTM

File: .github/actions/performance-tests/art.yaml [hunk 1]
```diff
@@ -26,8 +28,17 @@ scenarios:
           json:
             id: 1
             jsonrpc: "2.0"
-            method: "web3_clientVersion"
+            method: "web3_clientVersion"  # Deliberate typo to force failure
```
Comment (CharlieMc0) [inline .github/actions/performance-tests/art.yaml:31]: Is `# Deliberate typo to force failure` let over from testing?

**Review by CharlieMc0 — COMMENTED** [review]
I think there is one comment leftover from your testing but other than that it looks good.

File: .github/workflows/ci-nightly-performance-testing.yaml [hunk 2]
```diff
@@ -12,23 +12,57 @@ jobs:
...
-        if: success()
+        if: always()
         with:
           name: artillery-report
-          path: ./artillery_report.html
\ No newline at end of file
+          path: ./artillery_report.html
+
+      - name: "NOTIFY:SLACK:FAILURE"
+        if: failure()
```
Comment (gartnera) [inline .github/workflows/ci-nightly-performance-testing.yaml:62]: Please only send the alert if it's a ` schedule` type

**Review by gartnera — CHANGES_REQUESTED** [review]

**Review by gartnera — APPROVED** [review]

**Pull request merged**